resources/js/components/shippings/PaymentDetail.vue

```diff
@@ -1,128 +1,155 @@
+                </div> -->
+                <!-- </div> -->
+
+                <div style="border-bottom: 1px #b4b4b4 solid; margin: 10px 0px 10px 0px"></div>
+                Total Pembayaran
+                <div
+                    class="float-right"
+                    style="color: #ff8415"
+                >Rp {{formatPrice( merchant.totalProductCost + merchant.totalShippingCost) }}
+                </div>
+                <div>
+                    <span>Kurir</span>
+                    <span class="float-right font-weight-bold">{{ merchant.courier_used }}({{ merchant.estimate_waktu }})</span>
+                </div>
+                <div v-if="merchants.length > 1">
+                    <br>
+                </div>
+            </div>
+
+            <div class="mt-3">
+                <button class="btn essence-btn btn-block" @click.prevent="createOrder" :disabled="disable">Bayar
+                </button>
+            </div>
         </div>
-      </div>
-      <!-- {{ merchants }} -->
+        <!-- {{ merchants }} -->
     </div>
 
 </template>
 
 <script>
-import EventBus from "../../eventBus";
-
-export default {
-  props: ["userId"],
-  data() {
-    return {
-      merchants: [],
-      finalPaymentDetail: null,
-      disable: false,
-      cek : 0
+    import EventBus from "../../eventBus";
+
+    export default {
+        props: ["userId"],
+        data() {
+            return {
+                merchants: [],
+                finalPaymentDetail: null,
+                disable: false,
+                cek: 0,
+                kurir: []
+            };
+        },
+        methods: {
+            formatPrice(value) {
+                let val = (value / 1).toFixed().replace('.', ',')
+                return val.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ".")
+            },
+            registerListener() {
+                EventBus.$on("MERCHANT_LIST", merchants => {
+                    this.merchants = merchants;
+                    // console.log(this.merchants);
+                });
+
+                EventBus.$on("FINAL_TRANSACTION_DETAIL", finalPaymentDetail => {
+                    this.finalPaymentDetail = finalPaymentDetail;
+                });
+
+            },
+            changeKurir(id, idx) {
+                this.merchants[id].totalShippingCost = this.merchants[id].all_shipping[idx].cost_shipping.cost
+                this.merchants[id].estimate_waktu = this.merchants[id].all_shipping[idx].cost_shipping.etd
+                this.merchants[id].courier_code = this.merchants[id].all_shipping[idx].shipping_code
+                this.merchants[id].courier_used = this.merchants[id].all_shipping[idx].shipping_name
+                this.finalPaymentDetail.merchants[id] = this.merchants[id]
+                // console.log(this.finalPaymentDetail.merchants[id])
+            },
+            createOrder() {
+
+                //  this. registerListener();
+
+                if (this.cek === 1) {
```

Comment:
if yg ini gk mending di ganti aja? jadi if(this.check != 1 ){
this.disable = true
                    window.axios
                        .post("/api/transactions", this.finalPaymentDetail)
                        .then(res => {
                            if (this.finalPaymentDetail['merchants'].length == 1) {
                                window.location = "/customer/transactions/" + res.data.id
                            } else {
                                window.location = "/customer/" + this.userId + "/orders"
                            }
                        })
.............
......
}

Reply:
lw ku baca if nya itu gak ada gunanya sih kyk nya... karena udah di buat di button nya if else (bisa di submit atau engga)